### ISSUE TITLE
make cows blink

### ISSUE THREAD

**Issue by lstebner** [jeremyckahn/farmhand#94]
it would add a little bit of extra life to the cows, with minimal animation effort, if they sometimes blinked.

**Comment by jeremyckahn** [comment]
I like this idea! I think it would give the game some much-needed polish and personality.

In terms of implementation, I think it makes sense to forgo the perhaps more obvious GIF-based approach and instead do this programmatically. With GIFs, the blink cycle is consistent and unnatural (and possibly terrifying if all the cows blink at the same time). If we toggle between blink and non-blink states on random intervals with JavaScript, it'll look better and also cut down on the game's overall asset footprint.

**Comment by lstebner** [comment]
oh yeah! that would be creepy if they all did it at the same time haha. i was thinking the eyes could be removed from the main sprite, and they could have their own states/animation loops. so on some sort of randomish interval (with a min that is decently long), it would blink something like 1-3 times for that cow. each one would maintain their own randomish blink loop. but i feel like you're thinking of something even less graphical?

**Comment by jeremyckahn** [comment]
I won’t prescribe an implementation plan if I’m not the one implementing this, but if I do wind up being the implementer, here’s how I’d plan to do it:

  1. Create a “blinking state” and a “non-blinking state” image for each cow color, and name them all consistently
  2. In the cow animation code, randomly swap between the two at some reasonable random interval

This would require some experimentation (for instance, there may need to be a third mid-blink image), but it feels like a decent middle ground between simplicity and overall file size.

I think we have similar approaches in mind, especially with regard to animation timing. I think your strategy would be more efficient in terms of built bundle file size, but a little more complex to implement. Ultimately, whatever looks good, has good performance, and is maintainable is fine with me! 🙂

**Comment by lstebner** [comment]
yeah sounds like we're on the same page. your approach with a new image i think is a better MVP if you will. positioning the eyes was giving me mild nightmares if i'm honest. probably would have created some funny bugs 🤪
